### PR TITLE
feature - added production version bump & git tag push support

### PR DESCRIPTION
**Pull request by Amit-Biderman**
feature - added production version bump & git tag push support

### CODE REVIEW

File: .github/workflows/push.yml [hunk 1]
```diff
@@ -31,16 +31,17 @@ env:
...
-        go-version: '1.17'
     - name: calculate version
-      uses: ./
\ No newline at end of file
+      id: version
+      uses: ./
+    - name: Push version
+      run: |
+        git tag ${{ steps.version.outputs.tag }}
+        git push --tags
```
Comment (matan-chetz) [inline .github/workflows/push.yml]: we should not push tags here
Only push tags at the end of pipeline after testing and build are OK

File: pkgs/version/version.go [hunk 2]
```diff
@@ -20,8 +20,7 @@ func CheckRc(s string) bool {
 
 func GetLatestTag(environment string) (string, error){
 	log.Println("Fetching latest tag version...")
-	arg1 := "--sort='-*authordate' | head -n1"
-	out, err := exec.Command("sh", "-c", "git tag -l --merged", environment, arg1).Output()
+	out, err := exec.Command("sh", "-c", "git describe --tags $(git rev-list --tags --max-count=1)").Output()
```
Comment (matan-chetz) [inline pkgs/version/version.go:23]: Why did you change the git command ? what is the difference here ?

File: main.go [hunk 3]
```diff
@@ -83,10 +92,12 @@ func main() {
...
+		log.Println("Environment is staging")
 		finalTag := stagingVersion(commit.Tag, commit.Type, rc)
-		log.Println(finalTag)
-	case "main", "master":
-		finalTag := productionVersion(commit.Tag, rc)
-		log.Println(finalTag)
+		utils.SetTagOutputName(finalTag)
+	case "main", "master", "production":
+		log.Println("Environment is production")
+		finalTag := productionVersion(commit.Tag, commit.Type, rc)
```
Comment (matan-chetz) [inline main.go]: 1. for both staging version and production version you can commit as the first arg in the function this way you wont need to pass commit.Tag and commit.Type only commit for example
productionVersion(commit, rc)
2. In production Commit Type should be hard coded to path (bugfix)

**Review by matan-chetz — COMMENTED** [review]
Let me know if something is not clear

File: .github/workflows/push.yml (hunk 1 above)
Comment (Amit-Biderman) [reply .github/workflows/push.yml]: Currently the tests are not implemented, this is just a test pipeline. I can remove the push tags though.

File: pkgs/version/version.go (hunk 2 above)
Comment (Amit-Biderman) [reply pkgs/version/version.go:23]: Found a better way to get tags regardless or branches - when there were more than one tag in the old way had, i have to slice through a list and it became complicated. Using describe is the way to go

File: main.go (hunk 3 above)
Comment (Amit-Biderman) [reply main.go]: 1. - fixed
2. didn't know - fixed

File: pkgs/version/version.go (hunk 2 above)
Comment (matan-chetz) [reply pkgs/version/version.go:23]: remove environment param

File: pkgs/version/version.go [hunk 4]
```diff
@@ -30,14 +29,18 @@ func GetLatestTag(environment string) (string, error){
 }
 
 func TrimTag(latestTagRaw string) string {
-	log.Println("Trimming tag")
+	log.Println("Trimming tag:", latestTagRaw)
 	latest_tag := strings.Trim(latestTagRaw, "\n")
 	latest := strings.Trim(latest_tag, " ")
+	// After trimming - check if no previous tag exists, and return fallout tag
```
Comment (matan-chetz) [inline pkgs/version/version.go:35]: please add a log here tag tell that latest was not found and we used the default tag
something like that:
log.Println("Lastest tag was not found using default")
